GVRf/Framework/src/org/gearvrf/GVRSceneObject.java

```diff
@@ -698,6 +699,61 @@ public GVRSceneObject getChildByIndex(int index) {
      */
     public Iterable<GVRSceneObject> children() {
         return new Children(this);
+    }    
+
+    /**
+     * Add a {@link GVRBaseSensor} node to the object.
+     * 
+     * Note that the attached sensor affects this node and its descendants.
+     * 
+     * @param sensor
+     *            The {@link GVRBaseSensor} to be added to the object node.
+     *            
+     * @return <code>true</code> if the sensor was added successfully;
+     *         <code>false</code> otherwise, this could possibly be due to the
+     *         {@link GVRInputManager} not being set.
+     */
+    public boolean addSensor(GVRBaseSensor sensor) {
```

Comment:
Until a grouping node is added to GVRF, I'd say let's not hijack the GVRSceneObject to be more than what it is supposed to be. To me the lack of a grouping node is one more reason why this should be an extension library.

If a grouping node is added then you can do it the proper way where each sensor is its own node. If grouping node is never added we are not polluting a core geometry class but still have the sensor capability.